pkg/cluster/cluster.go

```diff
@@ -189,6 +190,10 @@ func (p *Peer) Info() map[string]interface{} {
 type PeerState struct {
 	Type    PeerType
 	APIAddr string
+
+	// Every peer can propagate its labels that makes a peer a unique member of the system.
+	// We assume that only Store Type peers will be propagating (external) labels.
+	Labels []storepb.Label
```

Comment:
I would not so much consider it as a uniqueness requirement. For sidecars it would ideally be – but no need to enforce/require it at this layer I believe. I'd just see it as a guarantee that all series exposed by this peer are guaranteed to have this label.

Then it should also not be a problem that store nodes have this as empty, since they provide no such guarantees.